rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -189,14 +212,19 @@ struct convert<rosbag2_storage::BagMetadata>
       node["compression_format"] = metadata.compression_format;
       node["compression_mode"] = metadata.compression_mode;
     }
+    if (metadata.version < 5) {
+      node["relative_file_paths"] = metadata.relative_file_paths;
+    }
+    if (metadata.version >= 5) {
```

Comment:
```suggestion
    } else {
```

Reply:
However, it's interesting that we even have versioning information here - `encode` should always be encoding the latest version, in this case 5. I would not be opposed to removing all versioning from `encode`.